js/lex.go

```diff
@@ -161,6 +161,9 @@ func (l *Lexer) Next() (TokenType, []byte) {
 	// ErrorToken, WhitespaceToken and CommentToken are already returned
 	if tt == LineTerminatorToken || tt == PunctuatorToken && regexpStateByte[c] {
 		l.regexpState = true
+	} else if tt == IdentifierToken {
+		hash := ToHash(l.r.Lexeme())
+		l.regexpState = hash != 0 && hash != This
```

Comment:
The fact is that the hashes can be expanded anytime to include more fast-lookups of strings. It is not meant to be only keywords, although there are mostly keywords now. I think it would be better to be explicit about which keywords would possibly trigger a regex match instead of assuming that no new hashes are added (this could introduce a seemingly weird bug in the future if a hash was added regarding regex matching).

I was thinking of something similar to https://github.com/tdewolff/minify/blob/master/html/table.go#L18 but then just with `true`'s. What do you think?